### PR TITLE
Add attach test

### PR DESCRIPTION
**Pull request by xlgao-zju**
Signed-off-by: Xianglin Gao <xlgao@zju.edu.cn>

### CODE REVIEW

File: pkg/validate/streaming.go [hunk 1]
```diff
@@ -18,7 +18,11 @@ package validate
 
 import (
 	"bytes"
+	"fmt"
+	// "io"
 	"net/url"
+	// "os"
```
Comment (feiskyer) [inline pkg/validate/streaming.go]: nit: remove unused imports

Comment (xlgao-zju) [inline pkg/validate/streaming.go]: yes, this pr is WIP. will update after we figure out why we can not get the stdout...

**Comment by xlgao-zju** [comment]
@feiskyer updated

**Comment by feiskyer** [comment]
Cool. Finally got this passed. LGTM.

**Comment by xlgao-zju** [comment]
@feiskyer Yes, it was painful. And thanks @Random-Liu for your suggestion about `ReadCloser`.

**Pull request merged**